Fix/Continue animation after click

tom-laventure opened the pull request:
Fix for #1404

tom-laventure commented:
Unsure how removing a single word from the changelog caused the tests to fail, I assume this is an issue with the general repo, let me know if not.

mattgperry commented on packages/framer-motion/src/value/index.ts:
```diff
@@ -419,6 +424,32 @@ export class MotionValue<V = any> {
         this.clearAnimation()
     }
 
+    /**
```
Rather than increase the API surface of `MotionValue` it would be preferable to just use the active animation directly.

mattgperry commented on packages/framer-motion/src/value/index.ts:
```diff
@@ -317,6 +317,11 @@ export class MotionValue<V = any> {
         return this.current
     }
 
+
```
Same as `play`/`pause` - preferable just to interface with the animation directly.

tom-laventure replied:
I've updated the code to access the animation directly, if you had another way in mind to accomplish this please let me know!

mattgperry commented:
Thanks for the PR! Added some comments

mattgperry commented:
Also this needs an e2e test - you can check out `dev/tests/drag.tsx` and the associated scripts in `packages/framer-motion/cypress/integration/drag.ts` for an example

tom-laventure commented on packages/framer-motion/src/animation/types.ts:
```diff
@@ -74,6 +74,7 @@ export type ElementOrSelector =
 export interface AnimationPlaybackControls {
     time: number
     speed: number
+    state?: string
```
Updated thanks!

tom-laventure commented on packages/framer-motion/src/gestures/pan/PanSession.ts:
```diff
@@ -143,14 +144,23 @@ export class PanSession {
         handlers: Partial<PanSessionHandlers>,
-        { transformPagePoint }: PanSessionOptions = {}
+        { transformPagePoint }: PanSessionOptions = {},
+        dragSnapToOrigin: boolean = false
```
Moved relevant code to `PanSessionOptions`

tom-laventure commented on packages/framer-motion/src/value/index.ts:
```diff
@@ -335,11 +340,11 @@ export class MotionValue<V = any> {
-                      parseFloat(this.prev as any),
-                  this.timeDelta
-              )
+            velocityPerSecond(
```
Prettier installed and resaved

tom-laventure commented:
Hi @mattgperry, updated the code with your requests and added an e2e test to the existing `drag-to-reorder` tests to check my changes are working. Let me know if you have any more changes or updates you would like implemented!

tom-laventure commented:
Hi @mattgperry, just bumping this for re-review, let me know if you have any thoughts.

Bonjour123 commented:
Any plan to merge soon ?

tomdaniels commented:
@mattgperry Hey mate any further considerations on this one that's potentially blocking it? If I was to fork from the original PR and get the CI build passing is this feasible to get into the library?
